### PR TITLE
[5 of 5: Configuration Updates] DCOS-5674: Update Configuration Progress Modal!

### PR DESCRIPTION
**Pull request by jfurrow**
Dependent PRs (this PR will be smaller when these merge):
#176 
#167 
#161 
#144

Also https://github.com/mesosphere/reactjs-components/pull/253 needs to merge before this works properly.

Still, sorry for the massive PR! Let me know if you need any help reviewing this!

I am writing integration tests now. :)

### CODE REVIEW

**Comment by mesosphere-web** [comment]
Reviewer: @pierlo-upitup

File: src/js/components/ServiceDetail.js [hunk 1]
```diff
@@ -22,8 +31,53 @@ class ServiceDetail extends mixin(InternalStorageMixin, TabsMixin) {
...
     this.state = {
-      currentTab: Object.keys(this.tabs_tabs).shift()
+      currentTab: Object.keys(this.tabs_tabs).shift(),
+      progressDetailModalOpen: false,
+      servicePlan: null
     };
+
+    this.store_listeners = [{
+      name: 'servicePlan',
+      events: [
```
Comment (mlunoe) [inline src/js/components/ServiceDetail.js]: Disclaimer: I don't see this in the previous PRs, so I'm going to comment here.
#nit Could we have this in one line?

File: src/js/components/ServiceDetail.js [hunk 2]
```diff
@@ -22,8 +31,53 @@ class ServiceDetail extends mixin(InternalStorageMixin, TabsMixin) {
...
+
+  onServicePlanStoreChange() {
+    this.setState({
+      servicePlan: ServicePlanStore.getServicePlan(this.props.service.id)
+    });
+  }
+
+  onServicePlanStoreError() {
+    this.setState({
+      servicePlan: null
```
Comment (mlunoe) [inline src/js/components/ServiceDetail.js]: Disclaimer: I don't see this in the previous PRs, so I'm going to comment here.
This could be in one line too

Comment (jfurrow) [inline src/js/components/ServiceDetail.js]: Done!

File: src/js/components/ServiceDetail.js (hunk 1 above)
Comment (jfurrow) [inline src/js/components/ServiceDetail.js]: Done!

**Comment by jfurrow** [comment]
Added integration tests ✅

**Comment by pierlo-upitup** [comment]
Kinda tough to review – ping me back once it's rebased.

**Comment by jfurrow** [comment]
Closing to split up into multiple PRs

**Pull request closed without merging**